### DISCUSSION TITLE
pypdfium2 publishment

### DISCUSSION THREAD

**Discussion by mara004** [pypdfium2-team/pypdfium2#10] [Q&A] [answered]
Do you think our PyPDFium2 repository is ready for being made public some time soon? Is there anything important that I missed?

My ambition would be to publish the stuff before the end of this year, perform a first upload to PyPI, and set up a readthedocs page.
Surely platform testing is not finished yet (#3), but I guess we won't get any further without help from others, so making the repository public could improve progress on testing.
If we publish PyPDFium2, I plan to archive the ancient pypdfium-reboot repository and its PyPI page, and ask the maintainer of the initial pypdfium to do so too. I haven't made a new release of pypdfium-reboot since I started working on PyPDFium2, and I think the current code base is already a lot better than the old one.

What's your opinion?

**Comment by mara004** [comment]
@adam-huganir I have started to set up readthedocs - could you please tell me your account name so I can add you to the contributors?

**Reply by adam-huganir** [reply]
`adam-huganir`

**Reply by mara004** [reply]
I've added you - not sure whether readthedocs sent you a notification or not.
The project dashbord is at https://readthedocs.org/projects/pypdfium2/

**Reply by mara004** [reply]
@YinlinHu If you could please tell me your rtd account name, I'll provide you with maintainer access, too.

**Comment by adam-huganir** [comment]
I don't see why not, though I would say just add a note to the README in the other repo that points people here rather than archiving. I feel like that is less common, and there are probably still people using `pypdfium` that won't have reason to upgrade yet so archiving that may have unintended consequences, though I can't think of anything off the top of my head

**Reply by mara004** [reply]
Archiving just makes it obvious that a repository is abandoned and that no further improvements will happen.
That's true at least for `pypdfium-reboot`. Of course archiving a repository doesn't force people to upgrade. Moreover, as far as I know, archiving is a revertable action and thus not really dangerous. The [GitHub docs](https://docs.github.com/en/repositories/archiving-a-github-repository/archiving-repositories) say: 
```
You can archive a repository to make it read-only for all users and indicate that it's no longer actively maintained. You can also unarchive repositories that have been archived.
```

**Reply by adam-huganir** [reply]
:+1:

**Reply by mara004** [reply]
However, your point of adding a note to the readme is also good - that may be a lot more informative than just archiving.

**Comment by adam-huganir** [comment]
@mara004 I'm in the middle of a pretty major refactor on the code that I have uses pypdfium and was going to give pypdfium2 a go in its place. Is there anything you want to do before you make this live on pypi?

**Reply by mara004** [reply]
> Is there anything you want to do before you make this live on pypi?

Basically no, at least I can't think of anything at the moment. We have a pretty usable support model API, documentation and a few tests. That should be fairly sufficient for an initial public release.
I would only have liked the Windows non-ascii filenames issue fixed upstream, but I'm not sure if/when that will happen, so we won't wait for it with our release.

**Comment by mara004** [comment, marked as answer]
I've published pypdfium2 now. pypdfium-reboot is archived; pypdfium includes a note referring to this repository.

Furthermore, I have submitted pull requests to all public projects using PyPDFium that I know of:
https://github.com/YinlinHu/kuafu/pull/1
https://github.com/innodatalabs/pdfbrain/pull/3 (superseded by different PR)
https://github.com/microsoft/OCR-Form-Tools/pull/1010
https://github.com/elescamilla/Extract-URLs/pull/1 (merged)
https://github.com/BlockSigner/wowpng/pull/4 (reclined as it's a demo repo only)

**Reply by mara004** [reply]
See https://github.com/pypdfium2-team/pypdfium2/discussions/134 for a more recent follow-up
